ompi/mca/coll/acoll/coll_acoll_bcast.c

```diff
@@ -223,100 +229,109 @@ static inline void coll_bcast_decision_fixed(int size, size_t total_dsize, int n
         }
     } else {
         if (acoll_module->use_dyn_rules) {
-            *sg_cnt = acoll_module->mnode_sg_size;
+            *sg_cnt = (acoll_module->mnode_sg_size == acoll_module->sg_cnt) ? acoll_module->sg_cnt : node_size;
             *use_0 = acoll_module->use_mnode;
             SET_BCAST_PARAMS(acoll_module->use_lin0, acoll_module->use_lin1, acoll_module->use_lin2)
         } else {
-            int derived_node_size = subc->derived_node_size;
             *use_0 = 1;
-            if (size <= (derived_node_size << 2)) {
-                size_t dsize_thresh[2][3] = {{512, 8192, 131072}, {128, 8192, 65536}};
-                int thr_ind = (size <= (derived_node_size << 1)) ? 0 : 1;
-                if (total_dsize <= dsize_thresh[thr_ind][0]) {
-                    *sg_cnt = node_size;
-                    SET_BCAST_PARAMS(0, 0, 0)
-                } else if (total_dsize <= dsize_thresh[thr_ind][1]) {
-                    *sg_cnt = sg_size;
-                    SET_BCAST_PARAMS(0, 0, 0)
-                } else if (total_dsize <= dsize_thresh[thr_ind][2]) {
-                    *sg_cnt = sg_size;
+            *sg_cnt = sg_size;
+            if (num_nodes == 2) {
```

Comment:
Minor comment: Can you please swap the arguments around  and have the constant first? i.e. this should be

```
if (2 == num_nodes) {
```
(this is just to follow the Open MPI coding guidelines). There are a number of similar instances in this routine, I will not point them out, but please apply the same rule throughout the patch. 

(Note: you are using this rule already at many other places)

Reply:
Done in all the relevant places in the latest update.